cdf_fabric_replicator/extractor.py

```diff
@@ -25,6 +30,35 @@ def __init__(self, stop_event: CancellationToken) -> None:
         self.azure_credential = DefaultAzureCredential()
         self.stop_event = stop_event
 
+    def run(self) -> None:
+        self.config = self.get_current_config()
+        self.client = self.config.cognite.get_cognite_client("cdf-fabric-extractor")
+        self.state_store = self.get_current_statestore()
+        self.state_store.initialize()
+
+        if not self.config.source:
+            self.logger.error("No source path or directory provided")
+            return
+
+        while self.stop_event.is_set() is False:
+            token = self.azure_credential.get_token("https://storage.azure.com/.default").token
```

Comment:
good point I can move it above the loop so it doesn't run every 5 sec.

Reply:
Depending on the workload in the loop (writing time series from Fabric to CDF, writing event from Fabric to CDF, downloading files from Fabric and uploading to CDF), as well as the duration of the services, we think this implementation can avoid the scenario where the service gets interrupted due to token expiration.

Reply:
Interesting thought. Maybe we should explicitly get credentials for a test identity that as has the right claim and scope for the test instead of calling `DefaultAzureCredential()` which might retrieve an existing token (https://learn.microsoft.com/en-us/python/api/azure-identity/azure.identity.defaultazurecredential?view=azure-python) and by that mask other issues in the code.

This dicussion on stackoverflow has some interesting thoughts around configuring logging when instantiating an indentity with `DefaultAzureCredential`.

Reply:
@xtophs  We dug deeper into the token expiry expectations and found that they expire at a random value between 60-90 minutes by default (https://learn.microsoft.com/en-us/entra/identity-platform/configurable-token-lifetimes#access-tokens). we introduced the get token back in the loop but check the time of expiration which is the only given property to indicate the token is expired.

Reply:
Also this is a good idea to look at in the future, but our recent change will address the timeout handling.

Reply:
agreed that this should be withi the loop for the extractor code.  Somehow i was still thnking about this being iin the tests.  It's A 595 line PR after all.

Reply:
Do we include additions and subtractions? for that metric?